src/python-common/ceph/deployment/drive_group.py

```diff
@@ -239,98 +239,90 @@ def _from_json_impl(cls, json_drive_group):
         :param json_drive_group: A valid json string with a Drive Group
                specification
         """
-        args: Dict[str, Any] = {}
+        args: Dict[str, Any] = json_drive_group.copy()
```

Comment:
I think this is why the tests are failing and also this prevents osds from actually creating from the UI. You are copying json_drive_group to the args here but the json_drive_group undergoes some change in the next lines and those changes are completely lost since the args never got updated. Simply moving this line after the if condition will fix this.

```suggestion

Reply:
looking...